### DISCUSSION TITLE
Is it possible to add HTML classes and custom stylesheets to the UI / themes as well?

### DISCUSSION THREAD

**Discussion by TonkatsuRa** [LilaRest/obsidian-custom-classes#14] [Q&A] [answered]
I am trying to find a way to use augmented-ui CSS library to style my theme. But for that I would need to find a plugin that can add to the HTML of the UI and load custom stylesheets.

Would that be even possible with the right plugin?

**Comment by LilaRest** [comment]
@TonkatsuRa I'm not sure to properly understand your question and why it is related to this Obsidian plugin.

But if you are writing your own Obsidian theme and want to import the augmented-ui library inside of it you could use the CSS `@import` directive at the top of your theme `styles.css` file:
```css
@import url('https://unpkg.com/augmented-ui@1.1.2/augmented.css');
```

**Reply by TonkatsuRa** [reply]
Hi LilaRest

Sorry if I wasn't clear. It isn't directly a question for this plugin per se. But I've seen that your plugin added custom classes. 
The problem with augmented-ui is, that importing it via CSS isn't enough for full functionality. I would somehow need to be able to attribute values to the HTML of <div> classes of the Obsidian UI itself (either through changing the HTML code or using JS). But my coding knowledge is rather limited and I am still learning, that's why I wanted to ask you (someone with more experience in Obsidian and HTML classes) if you see any way to achieve this?

![image](https://user-images.githubusercontent.com/115587388/236627432-b15ff815-d6af-4f14-8966-f6b65b325845.png)

**Reply by LilaRest** [reply, marked as answer]
@TonkatsuRa Ok got it ! 
Obsidian themes are only built from CSS files and no option is provided to mutate the DOM.
Which means that in order to append attribute to HTML elements you'll need to create an Obsidian plugin.

You could be inspired by the [Minimal](https://github.com/kepano/obsidian-minimal/tree/master) theme which has its [own plugin](https://github.com/kepano/obsidian-minimal-settings) to extend it with functionalities that are not available from pure CSS.

Also, the Obsidian team has just released [the developers documentation](https://docs.obsidian.md/Home) which may be a good place to start.

**Reply by TonkatsuRa** [reply]
I will give it a try but I fear, that I am way in over my head with this 😅

**Reply by LilaRest** [reply]
Good luck ! :blush:
